src/SFA.DAS.EmployerFinance/Commands/RefreshEmployerLevyData/LevyImportCleanupStrategy.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            var existingSubmissionIds = await _dasLevyRepository.GetEmployerDeclarationSubmissionIds(empRef);
+            var existingSubmissionIdsLookup = new HashSet<string>(existingSubmissionIds.Select(x => x.ToString()));
+
+            //NOTE: The submissionId in our database is the same as the declaration ID from HMRC (DasDeclaration)
+            declarations = declarations.Where(x => !existingSubmissionIdsLookup.Contains(x.Id)).ToArray();
+
+            declarations = declarations.Where(x => !DoesSubmissionPreDateTheLevy(x)).ToArray();
+
+            return declarations.Where(x => !IsSubmissionForFuturePeriod(x)).ToArray();
+        }
+        private bool DoesSubmissionPreDateTheLevy(DasDeclaration dasDeclaration)
+        {
+            return _hmrcDateService.DoesSubmissionPreDateLevy(dasDeclaration.PayrollYear);
+        }
+
+        private bool IsSubmissionForFuturePeriod(DasDeclaration dasDeclaration)
+        {
+            return dasDeclaration.PayrollMonth.HasValue && _hmrcDateService.IsSubmissionForFuturePeriod(dasDeclaration.PayrollYear, dasDeclaration.PayrollMonth.Value, DateTime.UtcNow);
+        }
+
+        private bool IsEndOfYearAdjustment(DasDeclaration dasDeclaration)
+        {
+            return dasDeclaration.PayrollMonth.HasValue && _hmrcDateService.IsSubmissionEndOfYearAdjustment(dasDeclaration.PayrollYear, dasDeclaration.PayrollMonth.Value, dasDeclaration.SubmissionDate);
+        }
+
+        private async Task UpdateEndOfYearAdjustment(string empRef, DasDeclaration yearEndAdjustment, DasDeclaration[] hmrcDeclarations)
+        {
+            if (yearEndAdjustment.LevyDueYtd == null && !yearEndAdjustment.NoPaymentForPeriod)
+            {
+                throw new ArgumentNullException(nameof(yearEndAdjustment));
+            }
+
+            yearEndAdjustment.EndOfYearAdjustment = true;
+
+            if (yearEndAdjustment.NoPaymentForPeriod)
+                return;
+
+            var period12Declaration = await GetDeclarationEffectiveForPeriod12(empRef, yearEndAdjustment.PayrollYear, yearEndAdjustment.SubmissionDate, hmrcDeclarations);
+
+            if (period12Declaration?.LevyDueYtd != null)
+            {
+                // Caution: the calculation here is incorrect (it should be the other way round). This results in negative values in LevyDeclaration EndOfYearAdjustmentAmount.
+                // However, do not fix this as somewhere later (probably in a view or sproc) the EndOfYearAdjustmentAmount is being inverted when writing to the transaction line table.
+                yearEndAdjustment.EndOfYearAdjustmentAmount =
+                    period12Declaration.LevyDueYtd.Value - yearEndAdjustment.LevyDueYtd ?? 0;
+            }
+            else
+            {
+                yearEndAdjustment.EndOfYearAdjustmentAmount = -yearEndAdjustment.LevyDueYtd ?? 0;
+            }
+        }
+
+        /// <summary>
```

Comment:
I'm not convinced on the need to include private method summary comments.  The majority of what is written should be obvious to the person reading and the methods and parameters are well enough named to give clarity.